### PR TITLE
Smoke test

### PR DESCRIPTION
**Pull request by hell0w0rld2**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Steps to Test
[ Describe how to run your AUTOMATED TESTS and any manual tests that may be need to prove your fix/feature ]
1.  Resets the estimated distance traveled
2. Drives the robot forwards about 100 inches
3. Runs the intake, hopper, and shooter motors in that order
4. Has the user hold up a red ball to the color sensor
5. Has the user break each of the break sensors

## Changelog

A brief description of the problem statement. 

- This is a smoke test to run through to see if each component on the robot is working. This is to be done before getting into the queue.

Explain the technical implementation of what you're doing.

- This could help the team figure out if any electrical components are broken before going into a match.

Explain why you're making this change and what benefits from it. 

- We would be able to see what, if anything, is broken before going into a match. An example of this would be the break sensors at the scrimmage.

## Automation
Issue: #23

### CODE REVIEW

File: autonomous/smokeTest.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        """Tests to see if the motors are working with an input from the driver"""
+        self.toDo = "Drives robot forwards until it reaches a certain distance"
+        self.driveTrain.setTank(-self.dumbSpeed, -self.dumbSpeed)
+        if int(self.driveTrain.getEstTotalDistTraveled()) >= 100 and int(self.driveTrain.getEstTotalDistTraveled()) <=115:
+            self.driveTrain.setTank(0, 0)
+            self.next_state("colorSensorCheck")
+        else:
+            self.next_state("drive")
+
+    """
```
Comment (ChrisHirsch) [inline autonomous/smokeTest.py:44]: Commented because it's not quite ready..more of an example

**Review by ChrisHirsch — DISMISSED** [review]
+1 👍 🥇

**Review by ManiacMoustache — CHANGES_REQUESTED** [review]
We should have log errors for when the checks pass as well, so if they have false positives we know.

**Review by ManiacMoustache — APPROVED** [review]
gg

**Review by ManiacMoustache — APPROVED** [review]
looks good

**Review by Leaf-life — APPROVED** [review]
good

**Review by tuffinmuffin — APPROVED** [review]

**Pull request merged**